### DISCUSSION TITLE
打包报错，没看懂是什么原因

### DISCUSSION THREAD

**Discussion by edifangyi** [umijs/umi#11225] [Q&A]
![image](https://github.com/umijs/umi/assets/12421552/facba092-79ba-45ee-89f7-07a83676ba89)

![image](https://github.com/umijs/umi/assets/12421552/5096d686-91d9-4217-9793-5e6378f08a51)
`  "dependencies": {
    "@ant-design/charts": "^1.4.2",
    "@ant-design/icons": "^4.8.0",
    "@ant-design/pro-card": "^1.20.16",
    "@ant-design/pro-descriptions": "^1.12.7",
    "@ant-design/pro-form": "^1.74.7",
    "@ant-design/pro-layout": "^6.38.22",
    "@ant-design/pro-list": "^1.21.87",
    "@ant-design/pro-table": "^2.80.8",
    "@icons/material": "^0.4.1",
    "@umijs/route-utils": "^2.0.3",
    "ali-oss": "^6.17.1",
    "antd": "^4.24.9",
    "classnames": "^2.3.0",
    "crypto-js": "^4.1.1",
    "lodash": "^4.17.0",
    "moment": "^2.29.4",
    "omit.js": "^2.0.2",
    "rc-menu": "^9.1.0",
    "rc-util": "^5.16.0",
    "react": "^17.0.0",
    "react-dev-inspector": "^1.7.0",
    "react-dom": "^17.0.0",
    "react-helmet-async": "^1.2.0",
    "umi": "^3.5.0"
  },`

可以运行得起来，但是打包报错，也不知道为啥

**Comment by fz6m** [comment]
umi 3 不维护了，建议升级到 umi 4

**Reply by coolMan2015** [reply]
umi4坑真多,还不如退回3

**Reply by tsuyoii** [reply]
同意，umi4用的真的感觉一言难尽，文档也找啥啥没有
